frontend/public/components/routes/create-route.tsx

```diff
@@ -0,0 +1,302 @@
+/* eslint-disable no-undef */
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import { Helmet } from 'react-helmet';
+import { Link } from 'react-router-dom';
+
+import { ButtonBar, Dropdown, history, resourcePathFromModel, ResourceName } from '../utils';
+import { k8sCreate, k8sGet, K8sResourceKind } from '../../module/k8s';
+import { formatNamespacedRouteForResource, getActiveNamespace } from '../../ui/ui-actions';
+import { ServiceModel, RouteModel } from '../../models';
+
+const getPortOptions = (service: K8sResourceKind) => {
+  if (!service) {
+    return {};
+  }
+
+  const ports = _.get(service, 'spec.ports', []);
+  const portOptions = ports.reduce((acc, { name, port, targetPort, protocol }) => {
+    acc[name] = <React.Fragment>{port} &rarr; {targetPort} ({protocol})</React.Fragment>;
+    return acc;
+  }, {});
+
+  return portOptions;
+};
+
+export class CreateRoute extends React.Component<null, CreateRouteState> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      formData: {
+        name: '',
+        hostname: '',
+        path: '',
+        service: '',
+        targetPort: '',
+        error: '',
+      },
+      toggleData: {
+        alternate: false,
+        secure: false,
+      },
+      loaded: false,
+      inProgress: false,
+      namespace: getActiveNamespace(),
+      services: [],
+      labels: {},
+      portOptions: {},
+    };
+  }
+
+  componentDidMount() {
+    k8sGet(ServiceModel, null, this.state.namespace)
```

Comment:
`k8sList` should be used for listing resource. It also returns the resource items by default https://github.com/openshift/console/blob/master/frontend/public/module/k8s/resource.js#L99